terragrunt/aws/api/lambda.tf

```diff
@@ -3,7 +3,6 @@ module "api" {
   name                     = "${var.product_name}-api"
   billing_tag_value        = var.billing_code
   allow_api_gateway_invoke = true
```

Comment:
You should be able to remove this as well - it'll clean up this element from the plan:
```hcl
resource "aws_lambda_permission" "api_gateway" 
```